### ISSUE TITLE
Application of subconv architecture in "real" tasks.

### ISSUE THREAD

**Issue by Sun694** [allenai/hidden-networks#2]
I was told by Mitchell Wortsman via email to open this issue:

I'm currently trying to apply the methods in the repo (https://github.com/tomgoldstein/loss-landscape) to a subconv network to look into the local loss minima of the model at the end of training. If possible, using the code in the above repo would simplify it, considering I would not have to re-implement it.

However, I ran into a problem when trying to use the code given by this repo: The traditional method of evaluating a (subconv network) seems to require a full set of arguments, which would require (at least) refactoring a lot of code for use in the above method, not to mention possibly having to include many helper functions.

Is there an existing easy way to turn a subconv model into a standalone model, free of the other model parameters? Is there an existing easy way to use this model in "real world" applications, free of most helper functions?

I could not find a turnkey solution present in the code of this repo, however, I fully admit my PyTorch is not up to par.

Thanks in advance for your time.

**Comment by mitchellnw** [comment]
Hi Jarad, sorry for the delay in responding to this issue.

Yes, unfortunately a turnkey solution does not exist in this repo but it shouldn't be too hard to construct something which resembled what you want. The `FixedSubnetConv` code here may be helpful for you: https://github.com/allenai/hidden-networks/blob/master/utils/conv_type.py#L125

If you use these instead of the regular `nn.Conv2d` then load a pretrained model and set the prune rate to something like 0.7 (corresponding to % weights = 30) you should be good! Basically, the `FixedSubnetConv` code looks at the scores and initializes a mask with the top k% of scores set as 1 and the rest to be 0. This mask is then fixed and the model is stand alone.

Hopefully this helps please, @vkramanuj do you have any additional suggestions here?

**Comment by vkramanuj** [comment]
The solution @mitchellnw provides above is probably the best for your use case, if you're using one of our pretrained models. Just be careful about using NonAffineBatchnorm (see utils/bntype.py) instead of LearnedBatchnorm when analyzing the loss landscape.
